### ISSUE TITLE
result.show() not displaying bounding boxes correctly

### ISSUE THREAD

**Issue by hienphan161** [mindee/doctr#1691]
### Bug description

I've discovered that using `straighten_pages=True` with images that are wider than they are tall and rotated by 90 or -90 degrees, appears to return incorrect bounding box coordinates. This is evidenced by `result.show()` not displaying the bounding boxes accurately.

Example input:
![vecteezy_modern-professional-business-card-creative-and-simple_9010481 - Copy](https://github.com/user-attachments/assets/b0490a31-a038-463e-b32f-83f5515067f2)
Output:
![download (5)](https://github.com/user-attachments/assets/f1a2f5b3-c63e-434f-b99b-f44533df279d)

### Code snippet to reproduce the bug

```
predictor = ocr_predictor(
  pretrained=True,
  assume_straight_pages=False,
  straighten_pages=True,
  detect_orientation=True,
  resolve_blocks=True
)

file_name = "/content/vecteezy_modern-professional-business-card-creative-and-simple_9010481 - Copy.jpg"
image = Image.open(file_name)
image = ImageOps.exif_transpose(image)

result = predictor([np.asarray(image)])
result.show()
```

### Error traceback

No error

### Environment

DocTR version: v0.9.0
TensorFlow version: 2.17.0
PyTorch version: 2.3.1+cu121 (torchvision 0.18.1+cu121)
OpenCV version: 4.10.0
OS: Ubuntu 22.04.3 LTS
Python version: 3.10.12
Is CUDA available (TensorFlow): Yes
Is CUDA available (PyTorch): Yes
CUDA runtime version: 12.2.140
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 535.104.05
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.9.6
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.9.6

### Deep Learning backend

is_tf_available: False
is_torch_available: True

**Comment by felixdittrich92** [comment]
Hi @hienphan161 :wave:;

Thanks for reporting, i was able to reproduce the issue.

It happens in the special case if we need to expand the image while correcting the rotation.
https://github.com/mindee/doctr/blob/d7f453329f583798c0d2774f343e29ab24450a4d/doctr/models/predictor/base.py#L105

The image is expanded correctly but the loc_preds aren't.

**Comment by hienphan161** [comment]
Thanks for looking into the issue so quickly.

I’m also curious why we only expand the image if it’s wider than it is tall. If we have an image that’s taller than it is wide and need to rotate it 90 or -90 degrees, wouldn’t we still need to expand the width of the image before rotation to avoid losing any part of it?

**Comment by felixdittrich92** [comment]
@hienphan161 fix on the way :)

**Comment by hienphan161** [comment]
Hi. I have another question about the 'straighten page' option. If I take a picture of a card at a 45-degree angle, like the one below, and I use the 'straighten page' option, the picture will be straightened perfectly. But, if the card isn't in the middle of the picture, we might end up cutting off some parts of the card, right?

Origin Image:
![20240822_154609](https://github.com/user-attachments/assets/fd715b04-2810-4b52-9534-fc61e1cdcf7a)

Rotated Image:
![f3ffb578f7cf17fc0689e61704b40aec83016ca1b1945587a65b86c2](https://github.com/user-attachments/assets/e9fe25f3-81c2-4810-bf3b-92e78c4f84db)

**Comment by felixT2K** [comment]
With the latest version v0.9.0. yes, but the fix is already merged (also linked here: https://github.com/mindee/doctr/pull/1697)
So available on the main branch (v0.9.1) or with the next release :)
